opencti-platform/opencti-front/src/private/components/settings/support/SupportPackageLines.tsx

```diff
@@ -0,0 +1,126 @@
+import SupportPackageLine from '@components/settings/support/SupportPackageLine';
+import usePreloadedPaginationFragment from '../../../../utils/hooks/usePreloadedPaginationFragment';
+import { UseLocalStorageHelpers } from '../../../../utils/hooks/useLocalStorage';
+import ListLinesContent from '../../../../components/list_lines/ListLinesContent';
+import { DataColumns } from '../../../../components/list_lines';
+
+const nbOfRowsToLoad = 50;
+
+interface SupportPackageLinesProps {
+  dataColumns: DataColumns;
+  queryRef: PreloadedQuery<SupportPackageLinesPaginationQuery>;
+  paginationOptions: SupportPackageLinesPaginationQuery$variables;
+  setNumberOfElements: UseLocalStorageHelpers['handleSetNumberOfElements'];
+}
+
+export const supportPackageLinesQuery = graphql`
+  query SupportPackageLinesPaginationQuery(
+    $search: String
+    $count: Int!
+    $cursor: ID
+    $orderBy: SupportPackageOrdering
+    $orderMode: OrderingMode
+  ) {
+    ...SupportPackageLines_data
+      @arguments(
+        search: $search
+        count: $count
+        cursor: $cursor   
+        orderBy: $orderBy
+        orderMode: $orderMode
+      )
+  }
+`;
+
+const supportPackageLinesFragment = graphql`
+  fragment SupportPackageLines_data on Query 
+  @argumentDefinitions(
+    search: { type: "String" }
+    count: { type: "Int", defaultValue: 25 }
+    cursor: { type: "ID" }
+    orderBy: { type: "SupportPackageOrdering", defaultValue: name }
+    orderMode: { type: "OrderingMode", defaultValue: asc } 
+  ) @refetchable(queryName: "SupportPackageLinesRefetchQuery") {
+    supportPackages(
+      search: $search
+      first: $count
+      after: $cursor
+      orderBy: $orderBy
+      orderMode: $orderMode
+    ) @connection(key: "Pagination_supportPackages") {
+      edges {
+        node {
+          id
+          ...SupportPackageLine_node
+        }
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+        globalCount
+      }
+    }
+  }
+`;
+
+const SupportPackageLines: FunctionComponent<SupportPackageLinesProps> = ({
+  queryRef,
+  setNumberOfElements,
+  dataColumns,
+  paginationOptions,
+}) => {
+  const { data, hasMore, loadMore, isLoadingMore } = usePreloadedPaginationFragment<
+  SupportPackageLinesPaginationQuery,
+  SupportPackageLines_data$key
+  >({
+    linesQuery: supportPackageLinesQuery,
+    linesFragment: supportPackageLinesFragment,
+    queryRef,
+    nodePath: ['supportPackages', 'edges'],
+    setNumberOfElements,
+  });
+
+  const queryData = usePreloadedQuery(supportPackageLinesQuery, queryRef);
+
+  const [_, refetch] = useRefetchableFragment<
+  SupportPackageLinesPaginationQuery,
+  SupportPackageLines_data$key
+  >(supportPackageLinesFragment, queryData);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      refetch({}, { fetchPolicy: 'store-and-network' });
+    }, 5000);
+
+    return () => clearInterval(intervalId);
+  }, [refetch]);
```

Comment:
Is this http polling a pattern we commonly use in the platform when listing items?

Reply:
it's used also in Data tab of a report for instance.
Not the best pattern I agree. Alternative is to use subscriptions.

Reply:
Changed to subscription.